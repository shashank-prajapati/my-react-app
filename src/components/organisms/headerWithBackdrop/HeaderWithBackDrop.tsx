import { Box, CircularProgress, Modal, Typography } from "@mui/material";
import Backdrop from "@mui/material/Backdrop";
import React, { useEffect } from "react";
import ExploreMenu from "../exploreMenu/ExploreMenu";
import Header from "../header/Header";
import TabLayout from "../tabLayout/TabLayout";
import { ReactComponent as ExpandMoreIcon } from "../../../media/expandMore.svg";
import { ReactComponent as ExpandLessIcon } from "../../../media/expandLess.svg";
import Icon from "../../atoms/icon/Icon";

type Props = {
  children?: React.ReactNode;
  onExploreMenuClick?: any;
  gotoMyLibrary?: any;
};

function HeaderWithBackDrop(props: Props) {
  const { children, onExploreMenuClick, gotoMyLibrary } = props;

  const [open, setOpen] = React.useState(false);
  const [icon, setIcon] = React.useState(
    <Icon
      iconComponent={ExpandMoreIcon}
      color="navTextColors.main"
      width="11px"
      height="6px"
    />
  );
  const handleOpen = () => {
    setOpen(!open);
    if (open) {
      setIcon(
        <Icon
          iconComponent={ExpandMoreIcon}
          color="navTextColors.main"
          width="11px"
          height="6px"
        />
      );
      document.body.style.overflow = "unset";
    } else {
      setIcon(
        <Icon
          iconComponent={ExpandLessIcon}
          color="navTextColors.main"
          width="11px"
          height="6px"
        />
      );
      document.body.style.overflow = "hidden";
    }
  };
  const handleClose = () => {
    setOpen(false);
    document.body.style.overflow = "unset";
    setIcon(
      <Icon
        iconComponent={ExpandMoreIcon}
        color="navTextColors.main"
        width="11px"
        height="6px"
      />
    );
  };

  return (
    <div>
      <Header
        btnClick={handleOpen}
        btnIcon={icon}
        gotoMyLibrary={gotoMyLibrary}
      />
      <Box id="content" sx={{ position: "relative", height: "100vh" }}>
        <Backdrop
          sx={{
            position: "absolute",
            color: "backDropColor.main",
            zIndex: (theme) => theme.zIndex.drawer + 1,
            height: "100vh",
          }}
          open={open}
          onClick={handleClose}
        >
          <Box sx={{ position: "absolute", top: 0 }}>
            <ExploreMenu onClick={onExploreMenuClick} />
          </Box>
        </Backdrop>
        {children}
      </Box>
    </div>
  );
}

export default HeaderWithBackDrop;
