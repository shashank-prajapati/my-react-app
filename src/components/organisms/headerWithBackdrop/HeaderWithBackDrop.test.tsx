import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import HeaderWithBackDrop from "./HeaderWithBackDrop";

describe("Testing Header with back drop", () => {
  it("should match the default snapshot", () => {
    const { container } = render(<HeaderWithBackDrop />);
    expect(container).toMatchSnapshot();
  });

  it("should open the dropdown", () => {
    const { container } = render(<HeaderWithBackDrop />);
    fireEvent.click(screen.getByText("Explore"));
    expect(screen.getByText("Entrepreneurship")).toBeInTheDocument();
  });
});
