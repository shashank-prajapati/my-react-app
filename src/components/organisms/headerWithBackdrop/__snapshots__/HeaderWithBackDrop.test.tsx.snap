// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header with back drop should match the default snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="MuiBox-root css-1ggpog0"
      >
        <div
          class="MuiBox-root css-70qvj9"
        >
          <img
            alt="Book"
            height="26px"
            src="blinkistLogo.png"
            style="display: block;"
            width="124px"
          />
          <div>
            <button
              aria-label="search button"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ig5hch-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                focusable="false"
              >
                searchIcon.svg
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-q987oy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Explore
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hpr88x-MuiSvgIcon-root"
                focusable="false"
              >
                expandMore.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-72xl0z-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            My Library
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-70qvj9"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-yw3ykx-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Signin
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-b9hkid"
      id="content"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-c7zf14-MuiBackdrop-root"
        style="opacity: 0; visibility: hidden;"
      >
        <div
          class="MuiBox-root css-1cq6gya"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1o8kjj3-MuiPaper-root"
          >
            <div
              class="MuiBox-root css-gg4vpm"
            >
              <span
                class="MuiTypography-root MuiTypography-subtitle3 css-cu8pq9-MuiTypography-root"
              >
                Explore by category
              </span>
              <span
                class="MuiTypography-root MuiTypography-subtitle3 css-1svm2tk-MuiTypography-root"
              >
                See recently added titles
              </span>
              <span
                class="MuiTypography-root MuiTypography-subtitle3 css-1svm2tk-MuiTypography-root"
              >
                See popular titles
              </span>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-ctpjxq-MuiDivider-root"
            />
            <ul
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              role="menu"
              tabindex="-1"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-jno0xv-MuiGrid-root"
                tabindex="0"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Vector.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Entrepreneurship
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Politics.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Politics
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Marketing Sales.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Marketing & Sales
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Science.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Science
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Health Nutrition.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Health & Nutrition
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Personal Development.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Personal Development
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Economics.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Economics
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        History.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        History
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Communication Skills.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Communication Skills
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Corporate Culture.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        CorporateCulture
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Motivation Inspiration.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Motivation & Inspiration
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Money Investments.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Money & Investments
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Psychology.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Psychology
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Productivity.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Productivity
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        SexRelationship.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Sex & Relationship
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Nature Environment.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Nature & Environment
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        CareerSuccess.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Career & Success
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                >
                  <div>
                    <div
                      class="MuiBox-root css-ccqql9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
                        focusable="false"
                      >
                        Education.svg
                      </svg>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-1ss91ko-MuiTypography-root"
                      >
                        Education
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
